dev/TeachingTip/TeachingTip.cpp

```diff
@@ -0,0 +1,1540 @@
+{
+    winrt::IControlProtected controlProtected{ *this };
+
+    m_rootGrid.set(GetTemplateChildT<winrt::Grid>(s_rootGridName, controlProtected));
+    m_nonBleedingContentRootGrid.set(GetTemplateChildT<winrt::Grid>(s_nonBleedingContentRootGridName, controlProtected));
+    m_bleedingImageContentPresenter.set(GetTemplateChildT<winrt::ContentPresenter>(s_bleedingImageContentPresenterName, controlProtected));
+    m_iconContentPresenter.set(GetTemplateChildT<winrt::ContentPresenter>(s_iconName, controlProtected));
+    m_actionButton.set(GetTemplateChildT<winrt::Button>(s_actionButtonName, controlProtected));
+    m_xCloseButton.set(GetTemplateChildT<winrt::Button>(s_xCloseButtonName, controlProtected));
+    m_closeButton.set(GetTemplateChildT<winrt::Button>(s_closeButtonName, controlProtected));
+    m_beakEdgeBorder.set(GetTemplateChildT<winrt::Grid>(s_beakEdgeBorderName, controlProtected));
+    if (SharedHelpers::IsThemeShadowAvailable())
+    {
+        m_shadowTarget.set(GetTemplateChildT<winrt::Grid>(s_shadowTargetName, controlProtected));
+        m_beakPolygon.set(GetTemplateChildT<winrt::Polygon>(s_beakPolygonName, controlProtected));
+    }
+
+
+    if (m_rootGrid)
+    {
+        m_ContentSizeChangedRevoker = m_rootGrid.get().SizeChanged(winrt::auto_revoke, {
+            [this](auto const&, auto const&)
+            {
+                UpdateSizeBasedTemplateSettings();
+                //Reset the currentEffectivePlacementMode so that the beak will be updated for the new size as well.
+                m_currentEffectivePlacementMode = winrt::TeachingTipPlacementMode::Auto;
+                TeachingTipTestHooks::NotifyEffectivePlacementChanged(*this);
+                if (IsOpen())
+                {
+                    PositionPopup();
+                }
+                if (m_expandAnimation)
+                {
+                    m_expandAnimation.get().SetScalarParameter(L"Width", static_cast<float>(m_rootGrid.get().ActualWidth()));
+                    m_expandAnimation.get().SetScalarParameter(L"Height", static_cast<float>(m_rootGrid.get().ActualHeight()));
+                }
+                if (m_contractAnimation)
+                {
+                    m_contractAnimation.get().SetScalarParameter(L"Width", static_cast<float>(m_rootGrid.get().ActualWidth()));
+                    m_contractAnimation.get().SetScalarParameter(L"Height", static_cast<float>(m_rootGrid.get().ActualHeight()));
+                }
+            }
+        });
+    }
+    if (m_closeButton)
+    {
+        m_closeButton.get().Click({ this, &TeachingTip::OnCloseButtonClicked });
+    }
+    if (m_xCloseButton)
+    {
+        m_xCloseButton.get().Click({ this, &TeachingTip::OnXCloseButtonClicked });
+    }
+
+    if (m_actionButton)
+    {
+        m_actionButton.get().Click({ this, &TeachingTip::OnActionButtonClicked });
+    }
+    UpdateButtonsState();
+    OnIconSourceChanged();
+    EstablishShadows();
+
+    if (Background())
```

Comment:
I think Luke already mentioned this, but this needs to be dynamic ... and also feels like something that should be handled by VSM? Or maybe just defaults in the template? Like can't the Style itself just have Background={ThemeResource yourdefault} and then if the developer sets it or restyles it then they win?